### ISSUE TITLE
Request: pip packages with CUDA 12 support

### ISSUE THREAD

**Issue by Talador12** [tensorflow/tensorflow#60943]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.12.0

### Custom Code

No

### OS Platform and Distribution

All

### Mobile device

All

### Python version

All

### Bazel version

All

### GCC/Compiler version

All

### CUDA/cuDNN version

12

### GPU model and memory

All

### Current Behaviour?

For reference, see the discussion on this PR and comment from @reedwm #58867 

```
This is a PR that has been merged, not an issue, so it cannot be reopened.

We have not yet released pip packages with CUDA 12 support, but are working on this. Feel free to file a new GitHub issue to have CUDA 12 pip packages (please CC me on the issue if you file it).
```

This pull request was focused on CUDA 12.X support in builds, but did not contain the scope for building new python packages for standard use on pypi
https://pypi.org/project/tensorflow/#history

Of note, Debian Bookworm is deprecating usage of CUDA 11.X versions, so only 12.X CUDA drivers are available. Tensorflow packages on pypi have yet to support CUDA 12.X drivers.

### Standalone code to reproduce the issue

```shell
See pypi, there is not a version of tensorflow with CUDA 12.X support
https://pypi.org/project/tensorflow/#history
```

### Relevant log output

_No response_</details>

**Comment by reedwm** [comment]
We are currently working on this, and hope to have pip packages with CUDA 12 soon.

**Comment by alanwilter** [comment]
Would this pip package encompass the CUDA/Nvidia packages dependency so it would work with GPUs straight out of the box, like `torch` package does?

**Comment by reedwm** [comment]
> Would this pip package encompass the CUDA/Nvidia packages dependency so it would work with GPUs straight out of the box, like `torch` package does?

@angerson do you know?

**Comment by angerson** [comment]
Not by default, no. @nluehr was working on https://github.com/tensorflow/tensorflow/pull/59825 so that a user could install `tensorflow` if they have CUDA already installed, or `tensorflow[and-cuda]` to install TensorFlow and the CUDA pip packages (which would then work with GPUs out of the box), but the PR has stalled recently.

**Comment by alanwilter** [comment]
> tensorflow[and-cuda]

That would be quite nice indeed.

**Comment by dcsouthwick** [comment]
Bumping and also wanted to ask if CUDA compute capability 9.0 by default will come with this update? JIT support is OK-ish, but waiting for a half an hour JIT compile every launch...

**Comment by nouiz** [comment]
@dcsouthwick In the mean time, there is some env variable that you can set to increase the JIT cache size to JIT compile only once: https://developer.nvidia.com/blog/cuda-pro-tip-understand-fat-binaries-jit-caching/

**Comment by marchcat69** [comment]
Any news?

**Comment by yasirjamal87** [comment]
Any updates here on when the fix is going to be made available?

**Comment by yasirjamal87** [comment]
Any updates?

**Comment by reedwm** [comment]
@reichlfl do you know the current status of upgrading the pip packages to CUDA 12? I see `tf.sysconfig.get_build_info()['cuda_version']` returns `'12.2'` in tf-nightly so perhaps this is already done in tf-nightly?

**Comment by nluehr** [comment]
Nightly moved to CUDA 12 a few days ago. See commit https://github.com/tensorflow/tensorflow/commit/3de44168950a5972ba4cfa7e3c6cbf4cffa67fe6.

**Comment by reedwm** [comment]
Closing this as fixed
